### PR TITLE
Add namespace-scoped collections with IVF vector indexing for agent database

### PR DESCRIPTION
**Pull request by JingYuChe**
## Summary by CodeRabbit

* **New Features**
  * Added namespace-enabled collections with namespace-scoped CRUD, querying, and `prewarm`, including stricter `use_namespace` routing and validation.
  * Introduced IVF dense vector indexing via `IVFConfiguration` (`IVFIndexType`/`IVFIndexLib`) and added `partition_count` + `has_vector_index` support.
  * Expanded schema/index configuration to accept IVF in addition to HNSW.
* **Bug Fixes**
  * Improved remote connection robustness and added more user-friendly namespace error messages.
* **Documentation**
  * Updated `query()` `include` to support `distances` and extended `where_document` docs with `$regex` (and updated logical operator guidance).
* **Tests**
  * Significantly expanded namespace/IVF/hybrid-search and lifecycle concurrency coverage, including prewarm and optional-index scenarios.

### CODE REVIEW

**Comment by JingYuChe** [comment]
@coderabbitai review

**Comment by JingYuChe** [comment]
@coderabbitai resume

**Comment by JingYuChe** [comment]
@coderabbitai review

**Review by hnwyllmm — COMMENTED** [review]
One API ergonomics issue: `create_collection(name, use_namespace=True)` currently appears to fail on the default path. When `schema` is omitted, `create_collection()` first builds the legacy default HNSW schema via `_prepare_schema_parameters()`, and then `_create_namespace_collection()` rejects it with `ValueError("use_namespace=True only supports IVF index type, HNSW is not allowed")`.

That error is technically accurate, but confusing for users because they did not explicitly choose HNSW; the SDK chose it as the default. For a namespace-enabled collection, the default path should either construct the namespace-compatible default schema, for example IVF plus the default fulltext config, or fail earlier with an actionable message such as: `use_namespace=True requires an IVF schema. The default non-namespace schema is HNSW. Please pass Schema(vector_index=VectorIndexConfig(ivf=IVFConfiguration(...))) or remove use_namespace=True.`

I would prefer making the default namespace path work if this is meant to match the Agent Database design shorthand, but at minimum the current error should explain why the default path became HNSW and how users should fix it.

**Review by hnwyllmm — COMMENTED** [review]
Another compatibility concern: namespace record IDs are currently validated with the same identifier-like pattern used for names (`[A-Za-z0-9_]`). That is stricter than the existing collection path, where `_id` supports arbitrary string formats and is safely escaped/cast before being written.

For record IDs, real applications may reasonably use UUIDs, file paths, URLs, external document IDs, or other opaque business keys, for example `repo/src/foo.py`, `https://example.com/doc/123`, or `550e8400-e29b-41d4-a716-446655440000`. These would work in a normal collection today but fail after moving to a namespace collection.

Could we relax namespace record-id validation to match the existing collection semantics, e.g. validate only type/non-empty/length and rely on parameterization/escaping plus JSON storage for safety? The stricter `[A-Za-z0-9_]` rule seems appropriate for resource names like collection/namespace/ltable names, but too restrictive for per-record business IDs.

**Review by hnwyllmm — COMMENTED** [review]
One more API-safety issue: `SparseVectorIndexConfig` appears to be silently ignored when creating a namespace-enabled collection.

The design doc says sparse vector support is out of scope for this phase, which is fine. But the public `Schema` still allows `sparse_vector_index=...`, and the namespace creation path does not reject it, persist it, or create a sparse index. That means a user can pass a sparse config, receive a successfully created collection, and only later discover that sparse retrieval is not available.

Could we explicitly reject `schema.sparse_vector_index is not None` when `use_namespace=True`, with a clear error such as `use_namespace=True does not support SparseVectorIndexConfig yet`? Silent feature drop is risky here, especially because the same schema works for normal collections.

**Review by hnwyllmm — COMMENTED** [review]
Test coverage is broad, but I think we still need a few public API compatibility tests for the namespace feature. Many current tests exercise validators, DDL fragments, include/n_results handling, or internal helpers such as `_create_namespace_collection()`. Those are useful, but they do not fully protect the real user-facing behavior.

Could we add tests around the public APIs, for example:

- `client.create_collection(name, use_namespace=True)` with no explicit schema: either succeeds with the intended namespace default schema, or fails with a clear actionable error.
- `client.get_or_create_collection(name, use_namespace=True)` when a normal collection with the same name already exists: should not silently return a non-namespace collection; it should raise a type/mode mismatch or otherwise make the behavior explicit.
- `client.create_collection(..., use_namespace=True, schema=...)` with unsupported configs such as HNSW or `SparseVectorIndexConfig`: should fail explicitly rather than silently ignoring unsupported pieces.
- Namespace record IDs should preserve ordinary collection compatibility, including IDs like UUIDs, paths, and URLs if we relax the validation.

These tests should go through `Client`/`BaseClient.create_collection()` and `get_or_create_collection()` rather than only calling internal methods, because the current regressions are mostly about public API composition and defaults.

**Review by hnwyllmm — COMMENTED** [review]
The integration coverage is impressive across lifecycle, DML, query, hybrid search, prewarm, drop, concurrency, and session variables. One gap I still see is compatibility and misuse coverage at the integration level.

Most namespace integration helpers appear to create collections with an explicit IVF schema, which is the happy path. Could we also add a small set of integration tests for edge/user-migration behavior, such as:

- Creating a namespace collection through the default public API path, or verifying the error is clear if default schema creation is intentionally unsupported.
- Attempting to reuse a normal collection name with `get_or_create_collection(..., use_namespace=True)` and asserting an explicit mode/type mismatch instead of returning the wrong collection type.
- Passing unsupported namespace schema options, such as HNSW or sparse vector index, and asserting they fail explicitly.
- Verifying namespace record IDs accept the same practical business-key shapes as normal collections, such as UUIDs, paths, and URLs, if validation is relaxed.
- If multi-LTable is part of the intended design scope, covering more than one logical table under the same namespace; if not, documenting/test-asserting the current single-default-LTable limitation.

These would complement the current happy-path namespace tests and protect backward compatibility for users migrating from normal collections.

**Review by hnwyllmm — COMMENTED** [review]
Could we also add user-facing documentation and example code for the namespace feature before merging?

This PR introduces a fairly large new API surface and some important constraints, so users will need a dedicated guide rather than only tests. I suggest adding a namespace guide that covers:

- Creating a namespace-enabled collection, including the required/recommended schema.
- Creating/getting/listing/deleting namespaces.
- Writing data with `add/update/upsert/delete` through the namespace API.
- Querying and `hybrid_search`, including `where`, `where_document`, `include`, and distances.
- `prewarm` behavior and deployment limitations.
- Deleting namespaces/collections and the async cleanup semantics if applicable.
- Compatibility and limitations: OceanBase/version requirement, unsupported schema options such as HNSW/sparse vector if they remain unsupported, record ID compatibility, and the current LTable behavior/scope.

A small runnable example under examples or docs would also help a lot, ideally showing the full flow from `create_collection(..., use_namespace=True)` to namespace CRUD/query and cleanup.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -95,7 +95,10 @@ jobs:
             log_level=INFO
           fi
           set -o pipefail
-          uv run pytest tests/integration_tests/ -v --log-cli-level=${log_level} -k "not server and not embedded and not oceanbase" | tee pytest.log
+          uv run pytest tests/integration_tests/ -v --log-cli-level=${log_level} \
```
Comment (hnwyllmm) [inline .github/workflows/ci.yml:98]: you shouldn't skip tests here. please skip the cases in the case files.
